### ISSUE TITLE
`stringFormatter.js` truncate non-ASCII characters unexpectedly

### ISSUE THREAD

**Issue by Max10240** [stylelint/stylelint#6860]
### What steps are needed to reproduce the bug?

To put it simply, this test case will fail (in `lib/formatters/__tests__/stringFormatter.test.js`):
```typescript
it('outputs warnings contains non-ASCII characters', () => {
		const results = [
			{
				source: 'path/to/file.css',
				warnings: [
					{
						line: 1,
						column: 1,
						rule: 'bar',
						severity: 'error',
						text: '简体中文こんにちは안녕하세요',
					},
				],
			},
		];

		const output = prepareFormatterOutput(results, stringFormatter);

		expect(output).toBe(stripIndent`
path/to/file.css
 1:1  ×  简体中文こんにちは안녕하세요  bar
1 problem (1 error, 0 warnings)`);
	});
```
in fact,  if a stylelint rule report error which `text` is '简体中文こんにちは안녕하세요', it will display '简体中文こんに'  inside the terminal(some characters are missing).

After locating it,  I found that the reason is:
the package `table` can't wrap non-ascii characters very well.
The smallest reproducible example:
```typescript
const table = require("table");

let config,
  data,
  output;

data = [
  ['简体中文'],
];

config = {
  columns: {
    0: {
      width: 20,
      wrapWord: true, // Note, enable `wrapWord`
    }
  }
};

output = table.table(data, config);

console.log(output); 
/*
╔══════════════════════╗
║ 简体                 
╚══════════════════════╝
*/
```
but if you disbale `wrapWord`, like: 
```typescript
// ...
config = {
  columns: {
    0: {
      width: 20,
      wrapWord: false, // Note, disable `wrapWord`
    }
  }
};
// ...
console.log(output); 
/*
╔══════════════════════╗
║ 简体中文             
╚══════════════════════╝
*/
```
it works well

### What Stylelint configuration is needed to reproduce the bug?

```typescript
{
  "rules": {
    "selector-class-pattern": [/foo/, { "message": '简体中文' }]
  }
}
```

```css
.bar {}
```

### How did you run Stylelint?

```shell
npx stylelint 'foo.css'
```

### Which version of Stylelint or dependencies are you using?

15.6.2

### What did you expect to happen?

output:
```shell
foo.css
 1:1  × 简体中文  selector-class-pattern

1 problem (1 error, 0 warnings)`);
```

### What actually happened?

output:
```shell
foo.css
 1:1  × 简体  selector-class-pattern

1 problem (1 error, 0 warnings)`);
```

note: '中文' is missing

### Does the bug relate to non-standard syntax?

No

### Proposal to fix the bug

disable `wrapWord` if error message contains non-ASCII characters.
if it is possible, i will open a PR for it later

**Comment by ybiquitous** [comment]
@Max10240 Thanks for the report with the reproducible example. Indeed, this is a bug.
